src/validator/LookupValidator.cpp

```diff
@@ -14,36 +14,55 @@ DECLARE_uint32(ft_request_retry_times);
 namespace nebula {
 namespace graph {
 
-/*static*/ constexpr char IndexScanValidator::kSrcVID[];
-/*static*/ constexpr char IndexScanValidator::kDstVID[];
-/*static*/ constexpr char IndexScanValidator::kRanking[];
+/*static*/ constexpr char LookupValidator::kSrcVID[];
+/*static*/ constexpr char LookupValidator::kDstVID[];
+/*static*/ constexpr char LookupValidator::kRanking[];
 
-/*static*/ constexpr char IndexScanValidator::kVertexID[];
+/*static*/ constexpr char LookupValidator::kVertexID[];
 
-Status IndexScanValidator::validateImpl() {
+Status LookupValidator::validateImpl() {
     NG_RETURN_IF_ERROR(prepareFrom());
     NG_RETURN_IF_ERROR(prepareYield());
     NG_RETURN_IF_ERROR(prepareFilter());
     return Status::OK();
 }
 
-Status IndexScanValidator::toPlan() {
-    auto *is = IndexScan::make(qctx_,
+Status LookupValidator::toPlan() {
+    auto* is = IndexScan::make(qctx_,
                                nullptr,
                                spaceId_,
                                std::move(contexts_),
                                std::move(returnCols_),
                                isEdge_,
                                schemaId_,
                                isEmptyResultSet_);
-    is->setColNames(std::move(colNames_));
-    root_ = is;
+    is->setColNames(std::move(idxScanColNames_));
+    PlanNode* current = is;
+
+    if (withProject_) {
+        auto* projectNode = Project::make(qctx_, current, newYieldColumns_);
+        projectNode->setInputVar(current->outputVar());
+        projectNode->setColNames(colNames_);
+        current = projectNode;
+    }
+
+    if (dedup_) {
+        auto* dedupNode = Dedup::make(qctx_, current);
+        dedupNode->setInputVar(current->outputVar());
+        dedupNode->setColNames(colNames_);
+        current = dedupNode;
+
+        // the framework will add data collect to collect the result
+        // if the result is required
+    }
```

Comment:
a question about which fields need to be de-dup？any field？

Reply:
The row in result.

Reply:
> The row in result.

got, If it's just rows de-dup, the graph layer doesn't need a separate node, a bool parameter is enough, It will be pushed down to the storage layer.
If any fields, the deduce node is required. for example :
lookup on t1 where ... yield distinct t1.c1, t1.c2
lookup on t1 where ... yield distinct _src, _dst

So we need to think about final design. That's why I want to remove `idxScanColNames_` of duplicate variable.